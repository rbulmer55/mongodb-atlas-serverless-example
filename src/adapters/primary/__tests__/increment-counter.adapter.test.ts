import * as incrementCounterUseCase from '@use-cases/counter/increment-counter.use-case';
import {incrementCounter} from '@adapters/primary/increment-counter.adapter';

let counter;

describe('increment-counter.adapter', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  beforeEach(() => {
    counter = {
      id: '12345',
      count: 3,
    };

    jest
      .spyOn(incrementCounterUseCase, 'incrementCounterUseCase')
      .mockResolvedValue(counter);
  });

  test('Adapter returns the counter on success', async () => {
    await expect(incrementCounter()).resolves.toMatchSnapshot();
  });
});
