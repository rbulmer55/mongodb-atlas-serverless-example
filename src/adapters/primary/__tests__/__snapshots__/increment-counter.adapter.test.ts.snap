// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`increment-counter.adapter Adapter returns the counter on success 1`] = `
{
  "count": 3,
  "id": "12345",
}
`;
